evaluator/src/main/kotlin/PackageRule.kt

```diff
@@ -70,6 +70,35 @@ open class PackageRule(
         licenseRules.forEach { it.evaluate() }
     }
 
+    /**
+     * A [RuleMatcher] that checks whether a severity of any vulnerability of the [package][pkg] using the specified
+     * [scoringSystem] is higher than [maxValue]. Tested using the [severityComparator]. Returns false if the severity
+     * is equal to [maxValue].
+     */
+    fun hasVulnerabilityWithSeverityOver(
+        maxValue: String,
+        scoringSystem: String,
+        severityComparator: Comparator<String>
+    ) =
+        object : RuleMatcher {
+            override val description =
+                "Higher severity than $maxValue defined by $scoringSystem found for ${pkg.id.toCoordinates()}."
+
+            override fun matches(): Boolean {
+                val advisorResults = ruleSet.ortResult.getAdvisorResults(true)
+                val resultsForPackage = advisorResults[pkg.id]
+
+                return resultsForPackage
```

Comment:
Yes it can be used.
While the merging of the vulnerabilities is not required for this functionality, I agree that `getVulnerabilites()` should be reused here.